crates/polars-ops/src/series/ops/cum_agg.rs

```diff
@@ -216,15 +216,38 @@ pub fn cum_max(s: &Series, reverse: bool) -> PolarsResult<Series> {
 }
 
 pub fn cum_count(s: &Series, reverse: bool) -> PolarsResult<Series> {
-    if reverse {
-        let ca: NoNull<UInt32Chunked> = (0u32..s.len() as u32).rev().collect();
-        let mut ca = ca.into_inner();
-        ca.rename(s.name());
-        Ok(ca.into_series())
-    } else {
-        let ca: NoNull<UInt32Chunked> = (0u32..s.len() as u32).collect();
-        let mut ca = ca.into_inner();
-        ca.rename(s.name());
-        Ok(ca.into_series())
+    // Fast paths for no nulls
+    if s.null_count() == 0 {
+        let out = cum_count_no_nulls(s.name(), s.len(), reverse);
+        return Ok(out);
     }
+
+    let mut count = 0 as IdxSize;
```

Comment:
Version with no reversals at all. Requires an extra variable but I assume this removes two reversal allocations.

```rust
pub fn cum_count(s: &Series, reverse: bool) -> PolarsResult<Series> {
    // Fast paths for no nulls
    if s.null_count() == 0 {
        let out = cum_count_no_nulls(s.name(), s.len(), reverse);
        return Ok(out);
    }

    let ca = s.is_not_null();
    let out: IdxCa = if reverse {
        let mut count = (s.len() - s.null_count()) as IdxSize;
        let mut decr: IdxSize = 0;
        ca.apply_values_generic(|v: bool| {
            count -= decr;
            decr = v as IdxSize;
            count
        })
    } else {
        let mut count = 0 as IdxSize;
        ca.apply_values_generic(|v: bool| {
            if v {
                count += 1;
            }
            count
        })
    };
    Ok(out.into())
}
```

Reply:
That is probably more efficient. Made a small adjustment but otherwise good input 👍